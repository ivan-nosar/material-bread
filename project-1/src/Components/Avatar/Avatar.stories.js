import React from 'react';
import { Image, View } from 'react-native';
import { storiesOf } from '../../storybook/helpers/storiesOf';

import { Avatar } from '../..';
import Header from '../../storybook/components/Header';
import Container from '../../storybook/components/Container';

/* eslint-disable no-console */

export default storiesOf('Components|Avatars', module)
  .addParameters({ jest: ['Avatar'] })
  .add('Image', () => (
    <Container>
      <Header title={'Image Avatar'} />
      <View style={{ flexDirection: 'row', flexWrap: 'wrap' }}>
        <Avatar
          type="image"
          image={
            <Image
              source={{
                uri:
                  'https://avatars1.githubusercontent.com/u/12564956?s=460&v=4',
              }}
            />
          }
          size={16}
        />
        <Avatar
          type="image"
          image={
            <Image
              source={{
                uri:
                  'https://avatars1.githubusercontent.com/u/12564956?s=460&v=4',
              }}
            />
          }
          size={24}
        />
        <Avatar
          type="image"
          image={
            <Image
              source={{
                uri:
                  'https://avatars1.githubusercontent.com/u/12564956?s=460&v=4',
              }}
            />
          }
          size={38}
          onPress={() => console.log('avatar')}
          ripple
        />
        <Avatar
          type="image"
          image={
            <Image
              source={{
                uri:
                  'https://avatars1.githubusercontent.com/u/12564956?s=460&v=4',
              }}
            />
          }
          size={56}
        />
        <Avatar
          type="image"
          image={
            <Image
              source={{
                uri:
                  'https://avatars1.githubusercontent.com/u/12564956?s=460&v=4',
              }}
            />
          }
          size={80}
        />
        <Avatar
          type="image"
          image={
            <Image
              source={{
                uri:
                  'https://avatars1.githubusercontent.com/u/12564956?s=460&v=4',
              }}
            />
          }
          size={70}
        />
      </View>
    </Container>
  ))
  .add('Icon', () => (
    <Container>
      <Header title={'Icon Avatar'} />
      <View style={{ flexDirection: 'row', flexWrap: 'wrap' }}>
        <Avatar
          type="icon"
          content="arrow-back"
          contentColor={'#c2185b'}
          color={'black'}
          size={16}
        />
        <Avatar type="icon" content="alarm" size={24} color={'#c2185b'} />
        <Avatar
          type="icon"
          content="add"
          contentColor={'#E91E63'}
          size={48}
          color={'#8BC34A'}
        />
        <Avatar
          type="icon"
          content="face"
          contentColor={'white'}
          size={64}
          color={'#42a5f5'}
        />
      </View>
    </Container>
  ))
  .add('Text', () => (
    <Container>
      <Header title={'Text Avatar'} />

      <Avatar
        type="text"
        content="CP"
        contentColor={'white'}
        size={64}
        color={'#42a5f5'}
      />
      <Avatar type="text" content="MB" size={48} color={'#c2185b'} />
      <Avatar
        type="text"
        content="NP"
        contentColor={'#c2185b'}
        color={'#f4511e'}
      />
    </Container>
  ));
