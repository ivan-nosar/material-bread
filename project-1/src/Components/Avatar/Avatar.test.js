import React from 'react';
import Avatar from './Avatar';

import renderer from 'react-test-renderer';

test('Avatar Renders', () => {
  const tree = renderer.create(<Avatar />).toJSON();
  expect(tree).toMatchSnapshot();
});
