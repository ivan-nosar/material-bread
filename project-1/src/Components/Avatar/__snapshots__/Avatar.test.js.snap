// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Renders 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#1e88e5",
        "borderRadius": 12,
        "height": 24,
        "justifyContent": "center",
        "width": 24,
      },
      undefined,
    ]
  }
/>
`;
